docs/API/core/app-actions-v2.md

```diff
@@ -0,0 +1,460 @@
+5. An app action (2nd gen) can run on server when `environment` property is set as `server` or `any`
+6. An app action (2nd gen) can run on client side when `environment` property is set as `client` or `any`
+7. An app action is **limited to 120 seconds of execution time**. After 120 seconds, the action will be killed and a specific timeout error will be returned and saved in the logs.
+8. The payload for on-demand actions is **limited to 2048 characters**.
+9. The result sent from an on-demand app action is **limited to 6MB**.
+10. Only JavaScript assets are loaded when the screen runs as an app action. Assets such as CSS and images will be ignored by the system.
+11. Scheduled app actions will only run the published version of a screen, whereas on-demand actions will run the version from the same environment they are fired from (e.g. Fliplet Viewer, Live apps  )
+
+---
+
+## Configuring function pipeline for app actions
+
+Our latest iteration of app actions require you to define a pipeline of functions to execute when the action runs. Additionally, functions can be nested into sub-pipelines.
+
+First, you want to fetch the array of functions available in the system:
+
+```js
+Fliplet.App.Tasks.Functions.get().then(function (listOfFunctions) {
+  // ...
+});
+```
+
+Each function has the following properties:
+
+- `id`: the Fliplet ID (formerly `widgetId`) of the function
+- `name`: the display name of the function
+- `package`: the package name of the function
+- `description`: the description of the function
+
+Here'a a sample code that creates a new app action with a pipeline of function.
+
+In the example, function `com.fliplet.function.if` function that behaves like an [if statement](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/if...else) and evaluates the contents of the input payload and checks if `foo` is `true`. If so, it will execute function `com.example.function.return-string` with the given settings. If `foo` is `false`, the function will be skipped.
+
+```js
+Fliplet.App.Actions.create({
+  name: 'sayHello',
+  functions: [
+    {
+      functionPackage: "com.fliplet.function.if",
+      settings: {
+        condition: "foo === true"
+      },
+      functions: [
+        { functionPackage: "com.example.function.return-string", settings: { message: "Hello world" } }
+      ]
+    }
+  ]
+})
+```
+
+Each function will be executed with the configured settings. An additional payload can be added when calling the action:
+
+```js
+Fliplet.App.Actions.run('sayHello', { foo: true });
+```
+
+### Action triggers
+
+An action can be triggered as a result of a system event (e.g. a data source being updated, a log entry being created). You can configure the triggers for an action using the `triggers` property.
+
+These are the available types of triggers:
+
+- `log`: Triggered when a log entry is created. <strong>Can be executed on server side only</strong>
```

Comment:
I think we should link to these docs here so users will understand what all the types are: https://developers.fliplet.com/Organization-audit-log-types.html#logs-from-fliplet-apps

Reply:
@debankurghosh2061 Updated accordingly